Feature: select image

neliudochka opened the pull request:
Added ability to select pictures from the gallery. Nothing more)
For the implementation used Intents.

P.S.
I updated android-studio, so IDE updated some files in the .idea directory and added a new one (.idea/kotlinc.xml). I read whether we need to add .idea directory to the GitHub repo at all and get the impression that yes and no at the same time. So, for now, I`ll leave them be, unless I find out the right result for this project.

Screenshots:
<img src="https://github.com/neliudochka/steganography/assets/90915275/54eae892-e63c-4771-8104-fc49606435d3" width=30% height=30%>
<img src="https://github.com/neliudochka/steganography/assets/90915275/22a23537-845e-4bd2-9d03-eb43eeec71ce" width=30% height=30%>
<img src="https://github.com/neliudochka/steganography/assets/90915275/2139198d-b66c-426c-9a76-13743290a865" width=30% height=30%>

AKushch1337 reviewed (changes requested):
- Add the opportunity to take photo directly from your application. You can have a look at this [article](https://medium.com/codex/how-to-use-the-android-activity-result-api-for-selecting-and-taking-images-5dbcc3e6324b) for inspiration
- Delete Shreck photos
- Fragments do not save their state: For example: when I load a photo, go to another section, then come back to the previous section, the uploaded photo won’t be there. It is necessary to either save the fragment state; or before moving to another section, if an image was loaded, ask the user if she/he really wants to move to another section and lose the entered data
- In EncodeResultFragment.kt delete unused import on line `import com.example.steganography.databinding.FragmentDecodeResultBinding
`

neliudochka commented:
- Rewritten image selection with Activity Result API part
Due to the fact, that using onActivityResult() is a bad  practice,
I rewrote the selection of the photo from the gallery with Activity Result API.

- Android Studio decided not to work, again. So this commit only exists because I reinstalled this beautiful IDE and some files, that are connected to the idea changed.

- I like your proposition about taking a photo directly from tht app, but it is not connected to the branch and pool request theme. I will implement this feature, but in the other branch.

- Deleted unused imports

- Deleted Shreck (default) photos
I don’t like your proposition, but I’ve to agree that it is better to delete these masterpieces, even though it is a national treasures. Nevertheless, I live them for now on the result layouts and will delete them, when will implement ResultFragments functionality.

neliudochka commented:
These changes take me so long because I can't solve problems connected with the Fragments saving state. Although, I've some ideas on how to make it work.
